### ISSUE TITLE
Kundera 2.8, Mongo, ManyToOne with MappedSuperClass

### ISSUE THREAD

**Issue by bsideup** [Impetus/kundera#435]
``` groovy
@Entity
@Table(name = "car_parts", schema = "NNS@NNSPU")
class CarPart {
    @Id
    @GeneratedValue
    String id;

    @ManyToOne
    CarPartResource visual;
}
```

``` groovy
@Entity
@Table(name = "car_part_resources", schema = "NNS@NNSPU")
class CarPartResource extends AbstractResource {

    @Column
    @Enumerated(EnumType.STRING)
    CarPartKind kind;

}
```

``` groovy
@MappedSuperclass
abstract class AbstractResource {
    @Id
    @GeneratedValue
    String id;

    @Column
    String uri;

    @ElementCollection
    Map<String,String> params;
}
```

this sample code fails with

```
Caused by InvalidEntityDefinitionException: com.cm.social.nns.model.entites.CarPartResource must have an @Id field.
->>  350 | validateIdAttribute in com.impetus.kundera.metadata.processor.TableProcessor
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
|    341 | validateAndSetId in     ''
|    153 | populateMetadata in     ''
|     87 | process      in     ''
|    131 | buildEntityMetadata in com.impetus.kundera.metadata.MetadataBuilder
|    391 | scanClassAndPutMetadata in com.impetus.kundera.configure.MetamodelConfiguration
|    230 | loadEntityMetadata in     ''
|    102 | configure    in     ''
|     65 | configure .  in com.impetus.kundera.configure.Configurator
|    109 | initializeKundera in com.impetus.kundera.KunderaPersistence
|     81 | createEntityManagerFactory in     ''
```

**Comment by bsideup** [comment]
https://github.com/impetus-opensource/Kundera/blob/9fe036b963dd7b8ef0ee55cfa24405284087e191/src/kundera-core/src/main/java/com/impetus/kundera/metadata/processor/TableProcessor.java?source=c#L338

here should be else block

**Comment by mevivs** [comment]
Fixed and releasing with 2.8.1
